packages/astro/src/@types/astro.ts

```diff
@@ -1010,6 +1012,71 @@ export interface AstroUserConfig {
 		 * ```
 		 */
 		service: ImageServiceConfig;
+
+		/**
+		 * @docs
+		 * @name image.domains (Experimental)
+		 * @type {Array<string>}
+		 * @default `{domains: []}`
+		 * @version 2.9.2
+		 * @description
+		 * Defines a list of domains allowed to be optimized locally
+		 *
+		 * The value should be an array of domain strings of images allowed to be optimized locally.
+		 * Wildcards are not permitted (see image.remotePatterns).
+		 *
+		 * ```js
+		 * {
+		 *   image: {
+		 *     // Example: Enable the Sharp-based image service
+		 *     domains: ['astro.build'],
+		 *   },
+		 * }
+		 * ```
+		 */
+		domains?: string[];
+
+		/**
+		 * @docs
+		 * @name image.remotePatterns (Experimental)
+		 * @type {Array<RemotePattern>}
+		 * @default `{remotePatterns: []}`
+		 * @version 2.9.2
+		 * @description
+		 * Defines a list of url patterns allowed to be optimized locally
```

Comment:
```suggestion
		 * Defines a list of permitted image source URL patterns for local image optimization.
```
Again, make this accurate, but I'm trying to hint at the idea of a remote image source. (The original descriptions of these didn't mention images at all, for example.)